embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/config/LocalConfigParser.kt

```diff
@@ -80,14 +75,35 @@ internal object LocalConfigParser {
         }
     }
 
+    private fun resolveAppId(
+        customAppId: String?,
+        resources: AndroidResourcesService,
+        packageName: String
+    ): String? {
+        return try {
+            customAppId ?: resources.getString(
+                resources.getIdentifier(
+                    BUILD_INFO_APP_ID,
+                    "string",
+                    packageName
+                )
+            )
+        } catch (exc: NotFoundException) {
+            null
+        }
+    }
+
     fun buildConfig(
         appId: String?,
         ndkEnabled: Boolean,
         sdkConfigs: String?,
         serializer: EmbraceSerializer,
         logger: EmbLogger
     ): LocalConfig {
-        require(!appId.isNullOrEmpty()) { "Embrace AppId cannot be null or empty." }
+        logger.logWarning(
+            "No appId supplied to Embrace. This is required if you want to " +
+                "send data to Embrace, unless you have configured an OTel exporter."
```

Comment:
can we detect if a user has configured an OTel exporter? if they haven't, i.e. they are sending data to our backend, it would probably be better if this threw an exception rather than issuing a warning log

Reply:
Maybe in debug? I'm wearying about a bad SDK configuration crashing the app if we threw in prod. Like passing a null into a Kotlin String or annotated @NonNull parameter, even though what they did isn't correct, I don't think we should do that.

Reply:
FWIW, an OTel exporter has to be configured before the SDK starts, so if we want to detect this case, we probably an.

Reply:
> Maybe in debug? I'm wearying about a bad SDK configuration crashing the app if we threw in prod.

yeah, fair point